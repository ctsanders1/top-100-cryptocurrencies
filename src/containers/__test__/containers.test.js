import React from 'react'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import colors from 'colors'
import ConnectedCurrencyFilter, { CurrencyFilter } from '../CurrencyFilter'
import ConnectedRefreshCurrenciesBtn, {
  RefreshCurrenciesBtn
} from '../RefreshCurrenciesBtn'

configure({ adapter: new Adapter() })

it('CurrencyFilter renders correctly', () => {
  const tree = renderer.create(<CurrencyFilter />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('ConnectedCurrencyFilter renders correctly', () => {
  const initiallState = {
      filter: 'USD',
      data: {},
      hasHttpError: { hasError: false, err: '' }
    },
    mockStore = configureStore(),
    store = mockStore(initiallState),
    wrapper = shallow(<ConnectedCurrencyFilter store={store} />),
    tree = renderer.create(wrapper).toJSON()

  expect(tree).toMatchSnapshot()

  console.log(
    'Test'.yellow,
    ' ConnectedCurrencyFilter renders correctly '.black.bgYellow,
    "is passing but throws warning  although CurrencyFilter component doesn't need key(s) - because it (component) doesn't render array of elements or components."
      .yellow
  )
})

it('RefreshCurrenciesBtn renders correctly', () => {
  const tree = renderer.create(<RefreshCurrenciesBtn />).toJSON()
  expect(tree).toMatchSnapshot()
})

it('ConnectedRefreshCurrenciesBtn renders correctly', () => {
  const initiallState = {
      filter: 'USD',
      data: {},
      hasHttpError: { hasError: false, err: '' }
    },
    mockStore = configureStore(),
    store = mockStore(initiallState),
    wrapper = shallow(<ConnectedRefreshCurrenciesBtn store={store} />),
    tree = renderer.create(wrapper).toJSON()

  expect(tree).toMatchSnapshot()
})
