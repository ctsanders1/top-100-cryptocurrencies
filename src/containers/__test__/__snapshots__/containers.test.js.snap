// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedCurrencyFilter renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    Change currency
  </div>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <input
        className="align-middle"
        defaultChecked={true}
        id="currencyUSD"
        name="currency"
        onClick={[Function]}
        type="radio"
        value="USD"
      />
      <label
        className="align-text-top ml-2"
        htmlFor="currencyUSD"
      >
         
        USD
      </label>
    </li>
    <li
      className="list-group-item"
    >
      <input
        defaultChecked={false}
        id="currencyEUR"
        name="currency"
        onClick={[Function]}
        type="radio"
        value="EUR"
      />
      <label
        className="align-text-top ml-2"
        htmlFor="currencyEUR"
      >
         
        EUR
      </label>
    </li>
    <li
      className="list-group-item"
    >
      <input
        defaultChecked={false}
        id="currencyCNY"
        name="currency"
        onClick={[Function]}
        type="radio"
        value="CNY"
      />
      <label
        className="align-text-top ml-2"
        htmlFor="currencyCNY"
      >
         
        CNY
      </label>
    </li>
  </ul>
</div>
`;

exports[`ConnectedRefreshCurrenciesBtn renders correctly 1`] = `
<button
  className="btn btn-outline-primary mb-5"
  onClick={[Function]}
>
  Get a fresh data
</button>
`;

exports[`CurrencyFilter renders correctly 1`] = `
<div
  className="card"
>
  <div
    className="card-header"
  >
    Change currency
  </div>
  <ul
    className="list-group list-group-flush"
  >
    <li
      className="list-group-item"
    >
      <input
        className="align-middle"
        defaultChecked={false}
        id="currencyUSD"
        name="currency"
        onClick={[Function]}
        type="radio"
        value="USD"
      />
      <label
        className="align-text-top ml-2"
        htmlFor="currencyUSD"
      >
         
        USD
      </label>
    </li>
    <li
      className="list-group-item"
    >
      <input
        defaultChecked={false}
        id="currencyEUR"
        name="currency"
        onClick={[Function]}
        type="radio"
        value="EUR"
      />
      <label
        className="align-text-top ml-2"
        htmlFor="currencyEUR"
      >
         
        EUR
      </label>
    </li>
    <li
      className="list-group-item"
    >
      <input
        defaultChecked={false}
        id="currencyCNY"
        name="currency"
        onClick={[Function]}
        type="radio"
        value="CNY"
      />
      <label
        className="align-text-top ml-2"
        htmlFor="currencyCNY"
      >
         
        CNY
      </label>
    </li>
  </ul>
</div>
`;

exports[`RefreshCurrenciesBtn renders correctly 1`] = `
<button
  className="btn btn-outline-primary mb-5"
  onClick={undefined}
>
  Get a fresh data
</button>
`;
